src/core/server/http/lifecycle/on_pre_response.ts

```diff
@@ -142,7 +142,11 @@ export function adoptToHapiOnPreResponseFormat(fn: OnPreResponseHandler, log: Lo
         if (preResponseResult.isNext(result)) {
           if (result.headers) {
             if (isBoom(response)) {
-              findHeadersIntersection(response.output.headers, result.headers, log);
+              findHeadersIntersection(
+                response.output.headers as { [key: string]: string },
```

Comment:
The new version of `@hapi/boom` ships with their own types. In these types they changed `output.headers` to `object`, whereas in `@types/hapi__boom` it's defined as `{[index: string]: string}`.

Either we use this hack and add `as { [key: string]: string }` to all places where we expect `ResponseHeaders`, or we update `ResponseHeaders` to allow `object`:

https://github.com/elastic/kibana/blob/008a420f81e10b3c98e2bdd2cca14feef0c61432/src/core/server/http/router/headers.ts#L58-L60